src/commands/SetPowerLevelCommand.ts

```diff
@@ -23,12 +23,39 @@ export async function execSetPowerLevelCommand(roomId: string, event: any, mjoln
     const level = Math.round(Number(parts[3]));
     const inRoom = parts[4];
 
+    const mjolnirId = await mjolnir.client.getUserId();
+
     let targetRooms = inRoom
         ? [await mjolnir.client.resolveRoom(inRoom)]
         : mjolnir.protectedRoomsTracker.getProtectedRooms();
 
+    let force = false;
+    if (parts[parts.length - 1] === "--force") {
+        force = true;
+        parts.pop();
+    }
+
     for (const targetRoomId of targetRooms) {
         try {
+            if (!force) {
+                if (target === mjolnirId || mjolnir.moderators.checkMembership(target)) {
```

Comment:
we should fail if `target === mjolnirId`, and `level < targetLevel`, even with `--force`

Reply:
wait, can you break that one down real quick? this sounds like you're saying it should fail when it tries to promote itself, which sounds like expected API behavior anyways?

Reply:
`targetLevel` is the current level - see other comment

Reply:
oh, you're trying to prevent demoting the bot itself? how would you handle gracefully removing the bot from a room, given you cant explicitly have it demote itself?

Reply:
you would puppet the bot and have it demote itself. Mjolnir is tolerable to a human operator using the account at the same time (deliberately - this is also why it doesn't filter itself out when checking for commands. Worst case, a human operator can go to the management room as the bot, send a ban command, and it all works fine)

Reply:
isn't the bot immune to self-bans via how powerlevels work? or is that not why protection-invoked selfbans fail?
besides, how does this work in an environment where humans dont possess either the login nor homeserver admin privilges, and use mjolnir to delegate permissions in an auditable fashion?